File: 559.maximum-depth-of-n-ary-tree.java
/*
 * @lc app=leetcode id=559 lang=java
 *
 * [559] Maximum Depth of N-ary Tree
 */
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
    public int maxDepth(Node root) {
        if (root == null) {
            return 0;
        }
        int max = 1;
        for (Node node : root.children) {
            max = Math.max(maxDepth(node) + 1, max);
        }
        return max;
    }
}

File: 559.maximum-depth-of-n-ary-tree.js
/*
 * @lc app=leetcode id=559 lang=javascript
 *
 * [559] Maximum Depth of N-ary Tree
 */
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number}
 */
var maxDepth = function (root) {
  let res = 0;
  if (!root) {
    return res;
  }
  

  let dfs = (root, depth)  => {
    if (root == null) {
      return;
    }
    res = Math.max(res, ++depth);
    for (let i = 0; i < root.children.length; i++) {
      dfs(root.children[i], depth);
    }
  }

  dfs(root, 0);

  return res;
};




